proxy/pkg/zdmproxy/coalescer.go

```diff
@@ -137,6 +141,13 @@ func (recv *writeCoalescer) RunWriteQueueLoop() {
 					}
 
 					log.Tracef("[%v] Writing %v on %v", recv.logPrefix, f.Header, connectionAddr)
+					if recv.frameProcessor != nil {
+						f = &frame.RawFrame{
+							Header: f.Header,
+							Body:   f.Body,
+						}
```

Comment:
I think it would be less error prone to move this to the frame processor itself and make the frame processor return an actual frame (the internal processor would return the same frame while the connector's frame processor would return a new frame with the same body but cloned header)

Reply:
I think I already commented something similar to this on the setstreamid methods but basically I think this should be moved to those 2 methods and then the new frame should be returned.

Reply:
And the stream id mapper implementation for the control connection doesn't have to clone anything imo, it can just return the existing frame.